### DISCUSSION TITLE
Long beatmaps cause huge lags in different scenarious

### DISCUSSION THREAD

**Discussion by Loreos7** [ppy/osu#19003] [Q&A]
### Type

Performance

### Bug description

Huge freeze while being selected in menu with a fps drop
Low fps when diving through scores

### Screenshots or videos

https://user-images.githubusercontent.com/86934170/177199740-2870c326-32eb-4b75-8a7a-f97366373d5e.mp4

https://user-images.githubusercontent.com/86934170/177201593-37048b49-db93-4379-a433-d5a6cffe1deb.mp4

### Version

704

### Logs

[database.log](https://github.com/ppy/osu/files/9041358/database.log)
[network.log](https://github.com/ppy/osu/files/9041359/network.log)
[performance.log](https://github.com/ppy/osu/files/9041360/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9041361/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9041362/updater.log)

**Comment by peppy** [comment]
Probably disk IO

**Comment by Loreos7** [comment]
I dont think so

![image](https://user-images.githubusercontent.com/86934170/177216957-ee3934f3-17fc-46bc-876b-9ee9a26d5a71.png)
![image](https://user-images.githubusercontent.com/86934170/177217011-d440907e-ea02-4b2a-9dec-28d0fc84bed3.png)

**Comment by Misadov** [comment]
I have same issue (NVME M2 SSD KINGSTON RBUSNS8154P3128GJ, so IO is fine I think). Moreover, there is no problem in the stable version of osu.

**Comment by peppy** [comment]
So it looks to be specific beatmaps. Could you please link the beatmaps in question?

**Reply by Loreos7** [reply]
https://osu.ppy.sh/beatmapsets/750284#taiko/1904970
https://osu.ppy.sh/beatmapsets/682489#mania/1443309
https://osu.ppy.sh/beatmapsets/162800#osu/1200519
https://osu.ppy.sh/beatmapsets/1249679#osu/2597284
https://osu.ppy.sh/beatmapsets/692689#taiko/2440314 - last diff

**Reply by Loreos7** [reply]
Could be the fact every single leaderboard has got 1 billion score. Just noticed

**Reply by peppy** [reply]
Can reproduce, will look into it.

**Reply by peppy** [reply]
I've identified three separate issues, all stemming from this beatmap being very long:

- Beatmap was loading synchronously (fixed via #19015)
- Audio track operations are running synchronously, which means that a track loading in the background can unintentionally stall unrelated `Stop` or `Start` calls. I've fixed this via https://github.com/ppy/osu-framework/pull/5294 / #19026  but may need further discussion as I'm not sure how acceptable this solution is.
- Difficulty calculation incurs huge overheads on other threads, which even with enough cores can be felt through blocking GC operations. This can be reproduced by rapidly switching back and forth to the beatmap. The CPU and allocation overhead is stemming from bindable allocations and initial setter operations (samples are a large one in the case of the first mania beatmap). I'll do a pass on this today to see if there are any quick gains, as it's pretty hefty.

**Reply by Loreos7** [reply]
https://user-images.githubusercontent.com/86934170/177526028-7026cc3a-8c4c-4d5e-bbe2-403ec6419ecc.mp4

And after 3 minutes of waiting 

https://user-images.githubusercontent.com/86934170/177526607-3153836c-9b4e-4270-9af0-654d28699e64.mp4

**Reply by peppy** [reply]
The editor is out of scope for such beatmaps at the moment.

**Comment by Loreos7** [comment]
https://user-images.githubusercontent.com/86934170/178090404-18db01a5-7f14-4b3b-8805-dab07b750e0c.mp4

And then even my OBS said: "An unexpected error has occured while recording"

**Reply by peppy** [reply]
Why is this video in this thread? Please open a new discussion and actually post text describing your new issue, thanks.
